core/src/test/scala/unit/kafka/log/LogCleanerManagerTest.scala

```diff
@@ -571,12 +571,12 @@ class LogCleanerManagerTest extends Logging {
     cleanerManager.setCleaningState(topicPartition, LogCleaningInProgress)
     cleanerManager.doneCleaning(topicPartition, log.dir, 1)
     assertTrue(cleanerManager.cleaningState(topicPartition).isEmpty)
-    assertTrue(cleanerManager.allCleanerCheckpoints.get(topicPartition).nonEmpty)
+    assertTrue(cleanerManager.allCleanerCheckpoints.contains(topicPartition))
```

Comment:
I think we could make the assertion a little stronger and fix this at the same time. We expect the checkpoint to be exactly the `endOffset` provided in `doneCleaning`. So maybe we could do something like this?
```scala
    val endOffset = 1L
    cleanerManager.doneCleaning(topicPartition, log.dir, endOffset)
    assertTrue(cleanerManager.cleaningState(topicPartition).isEmpty)
    assertEquals(Some(endOffset), cleanerManager.allCleanerCheckpoints.get(topicPartition))
```

Reply:
+1